### PR TITLE
Added support for configurable (up to 5) cut points for asmts.

### PR DESCRIPTION
**Pull request by agorina**

### CODE REVIEW

File: package-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultAssessmentParser.java [hunk 1]
```diff
@@ -142,27 +154,96 @@ void parse(final CSVRecord record) {
...
+            if (IAB.id() == asmtTypeId) {
+                parseIABScores(record, asmtBuilder);
+            } else {
+                parseNonIABScores(record, asmtBuilder);
+            }
+        }
+
+        /**
+         * SBAC IABs do not play by the same rules as ICA/Summatives
+         */
```
Comment (agorina) [inline package-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultAssessmentParser.java]: I have created https://jira.fairwaytech.com/browse/DWR-2047 to add some validation to the subject ingest.

File: migrate-reporting/src/test/resources/WarehouseEntitiesSetup.sql [hunk 2]
```diff
@@ -205,12 +205,12 @@ INSERT INTO asmt_target_exclusion(asmt_id, target_id) VALUES
...
 
-INSERT INTO asmt_score (asmt_id, cut_point_1, cut_point_2, cut_point_3, min_score, max_score) VALUES
-  (-99, 2442, 2502, 2582, 2201, 2701),
-  (-98, 2442, 2502, 2582, 2201, 2701),
-  (-59, 2442, 2502, 2582, 2201, 2701),
-  (-111, 2442, 2502, 2582, 2201, 2701),
-  (-311, 2442, 2502, 2582, 2201, 2701);
+INSERT INTO asmt_score (asmt_id, cut_point_1, cut_point_2, cut_point_3, cut_point_4, cut_point_5, min_score, max_score) VALUES
+  (-99,  2442, 2502, 2582, null, null, 2201, 2701),
+  (-98,  2442, 2502, 2582, 2590, 2595, 2201, 2701),
```
Comment (malaffoon) [inline migrate-reporting/src/test/resources/WarehouseEntitiesSetup.sql]: That's some itty-bitty performance levels!  ;)

File: package-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultAssessmentParser.java (hunk 1 above)
Comment (malaffoon) [inline package-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultAssessmentParser.java]: Perhaps more in this comment?  `Specifically, it is expected that there are 4 cut-points provided. The first and last are used to get the min/max score and the the second one (between levels 2 and 3) is used for the per-student performance level calculation.`

File: package-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultAssessmentParser.java [hunk 3]
```diff
@@ -142,27 +154,96 @@ void parse(final CSVRecord record) {
...
+        private void parseNonIABScores(final CSVRecord record, Builder asmtBuilder) {
+            final Integer performanceLevelCount = getPerformanceLevelCount();
+            if (performanceLevelCount == null) return;
+
+            asmtBuilder.minScore(toScore(ScaledLow1, record));
+
+            final String maxScoreColHeader = scaleColScoreHeaders.get(performanceLevelCount - 1);
+            asmtBuilder.maxScore(toScore(maxScoreColHeader, record));
+
+            for (int i = 0; i <= performanceLevelCount - 2; i++) {
```
Comment (malaffoon) [inline package-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultAssessmentParser.java]: I don't know if it is better but i wonder if it is more readable to do something like:
```
    if (performanceLevelCount > 1) asmtBuilder.cutPoint1(toScore(ScaledHigh1, record));
    if (performanceLevelCount > 2) asmtBuilder.cutPoint2(toScore(ScaledHigh2, record));
    if (performanceLevelCount > 3) asmtBuilder.cutPoint2(toScore(ScaledHigh3, record));
    if (performanceLevelCount > 4) asmtBuilder.cutPoint2(toScore(ScaledHigh4, record));
    if (performanceLevelCount > 5) asmtBuilder.cutPoint2(toScore(ScaledHigh5, record));
```

**Pull request merged**